### ISSUE TITLE
Standard error of Scala.js is erroneously sent to `logger.error`, instead of to stderr

### ISSUE THREAD

**Issue by cuzfrog** [scala-js/scala-js#3206]
Use nodejs `child_process` to execute command:
```scala
ChildProcess.execSync("npm test") //trigger jest
```
output in sbt console:
```
[error] PASS js-tests/target/tmp.test.js
[error]   ✓ adds 1 + 2 to equal 3 (4ms)
[error] 
[error] Test Suites: 1 passed, 1 total
[error] Tests:       1 passed, 1 total
[error] Snapshots:   0 total
[error] Time:        0.6s, estimated 1s
[error] Ran all test suites.
```

`ChildProcess` definition:
```scala
@js.native
@JSImport("child_process", JSImport.Namespace)
object ChildProcess extends js.Object {
  def execSync(command: String, options: js.UndefOr[js.Object] = js.undefined): js.Object | String = js.native
}
```

It's favorable to allow clients to give control to this logger.

workaround:
`child_process.spawnSync(command[, args][, options])` returns an object containing instead of printing the output.

**Comment by sjrd** [comment]
The important bit is that you said on gitter that the JVM version of the same code does not have the `[error]` prefixes. Could you also clarify that in the issue description, along with the code that produces the behavior?

**Comment by cuzfrog** [comment]
On jvm, use `scala.sys.process.Process` to run the same command, sbt console looks like:
```
PASS js-tests/target/tmp.test.js
  ✓ adds 1 + 2 to equal 3 (4ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        0.592s, estimated 1s
Ran all test suites.
```

**Comment by sjrd** [comment]
Minimized down to:
```scala
package helloworld

object HelloWorld {
  def main(args: Array[String]): Unit = {
    System.out.println("on stdout")
    System.err.println("on stderr")
  }
}
```
In a JVM project, `helloworld/run` displays
```
on stdout
on stderr
```
but in a JS project, it displays
```
on stdout
[error] on stderr
```

**Comment by sjrd** [comment]
Fun fact: the various JS envs don't even agree on what to do with the standard error. Node.js and Node.js+jsdom send the standard error to `logger.error`, but the other environments (PhantomJS and Rhino) send it to `stdout`.

For PhantomJS, that's mostly because PhantomJS itself does not have a separate channel for `console.error` (https://github.com/ariya/phantomjs/issues/10150), so there's nothing we can do about that one.

For Rhino, that's because the Rhino env only declares `console.log` and not `console.error` in the script's environment, so our `System.err.println` falls back on `console.log`.

Since Rhino is going away, and PhantomJS is a lost cause, I would only try to fix Node.js on this one. Moreover, a proper fix can only be done in 1.x in the context of #3033. So maybe in 0.6.x all we should do is unify Node.js with the other environments, and send `console.error`s from the script to `jsConsole.log` as well (so basically to stdout).

**Comment by sjrd** [comment]
Mostly fixed in bb9c6cbee2e5e93bf6ac5f020d74f98444eb009b: the standard error will now be redirected to stdout instead of the `logger.error`. As a further improvement, it should be sent to stderr, but that can only be done in Scala.js 1.x, and is already part of the job description in #3033.

So I'm going to close this issue.
